web/pages/reports.tsx

```diff
@@ -0,0 +1,297 @@
+import { useEffect, useState, useRef } from 'react'
+import { Col } from 'web/components/layout/col'
+import { Page } from 'web/components/layout/page'
+import { Row } from 'web/components/layout/row'
+import { QueryUncontrolledTabs } from 'web/components/layout/tabs'
+import { RelativeTimestamp } from 'web/components/relative-timestamp'
+import { SEO } from 'web/components/SEO'
+import { UserHovercard } from 'web/components/user/user-hovercard'
+import { Avatar } from 'web/components/widgets/avatar'
+import { ChoicesToggleGroup } from 'web/components/widgets/choices-toggle-group'
+import { Title } from 'web/components/widgets/title'
+import { BannedBadge, UserLink } from 'web/components/widgets/user-link'
+import { useAdminOrMod } from 'web/hooks/use-admin'
+import { getUserById } from 'web/lib/supabase/users'
+import { api } from 'web/lib/firebase/api'
+import { Report, ReportStatus } from 'common/api/report-types'
+import Link from 'next/link'
+import { EditableModNote } from 'web/components/contract/editable-mod-note'
+
+const updateReport = async (
+  reportId: number,
+  updates: Partial<{ status: ReportStatus; mod_note: string }>
+) => {
+  const response = await api('update-report', { reportId, updates })
+  if (response.status === 'success') {
+    return response.data
+  } else {
+    console.error('Error updating report:', response)
+    return null
+  }
+}
+
+const App = () => {
+  const isAdminOrMod = useAdminOrMod()
+  const [reports, setReports] = useState<Report[]>([])
+  const [initialLoading, setInitialLoading] = useState(true)
+  const [owner, setOwner] = useState<{ [key: string]: any }>({})
+  const [reportStatuses, setReportStatuses] = useState<{
+    [key: number]: ReportStatus
+  }>({})
+  const [modNotes, setModNotes] = useState<{
+    [key: number]: string | undefined
+  }>({})
+  const initialLoadRef = useRef(true)
```

Comment:
Don't use refs for this. Mutating a ref does not trigger a re-render.

Most likely `reports` should optionally be `undefined`, and default to undefined, and then you know when it's loaded when it becomes defined. Then you don't need another piece of state.